src/domain/entities/Template.ts

```diff
@@ -356,3 +358,178 @@ export function getDataSources(template: Template, sheetName: string): DataSourc
+                coordinate: undefined,
+                trackedEntityInstance: undefined,
+                programStage: undefined,
+                dataValues: entry.dataValues.map((dv: DataSetPackageDataValue) => ({
+                    dataElement: dv.dataElement,
+                    value: dv.value,
+                    category: dv.category,
+                    optionId: undefined,
+                })),
+            })),
+        };
+    }
+
+    if (dataPackage.type === dataFormTypeMap.programs || dataPackage.type === dataFormTypeMap.trackerPrograms) {
+        const mappedEntries: TemplateDataPackageData[] = dataPackage.dataEntries.map((entry: ProgramPackageData) => ({
+            group: undefined,
+            dataForm: entry.dataForm,
+            id: entry.id,
+            orgUnit: entry.orgUnit,
+            period: entry.period,
+            attribute: entry.attribute,
+            coordinate: entry.coordinate ?? convertCoordinate(entry.geometry ?? undefined),
+            trackedEntityInstance: entry.trackedEntityInstance,
+            programStage: entry.programStage,
+            dataValues: entry.dataValues.map(dv => ({
+                dataElement: dv.dataElement,
+                value: dv.value,
+                category: undefined,
+                optionId: undefined,
+            })),
+        }));
+
+        if (dataPackage.type === dataFormTypeMap.trackerPrograms) {
+            return {
+                ...dataPackage,
+                dataEntries: mappedEntries,
+                trackedEntityInstances: dataPackage.trackedEntityInstances,
+            };
+        }
+
+        return {
+            type: "programs",
+            dataEntries: mappedEntries,
+        };
+    }
+
+    throw new Error("Unknown data package type");
+}
+
+function convertToGeometry(entry: TemplateDataPackageData): Maybe<Geometry> {
+    const coord = entry.coordinate;
+    if (!coord) return undefined;
+
+    const longitude = Number(coord.longitude);
+    const latitude = Number(coord.latitude);
+
+    return {
+        type: "Point",
+        coordinates: [longitude, latitude],
+    };
+}
+
+function mapProgramEntry(entry: TemplateDataPackageData): ProgramPackageData {
+    return {
+        orgUnit: entry.orgUnit,
+        dataForm: entry.dataForm,
+        period: entry.period,
+        attribute: entry.attribute,
+        id: entry.id,
+        trackedEntityInstance: entry.trackedEntityInstance,
+        programStage: entry.programStage,
+        coordinate: entry.coordinate,
+        geometry: convertToGeometry(entry),
+        dataValues: entry.dataValues.map(dv => ({
+            dataElement: dv.dataElement,
+            value: dv.value,
+        })),
+    };
+}
+
+function convertCoordinate(geometry?: Geometry): Maybe<TemplateDataPackageData["coordinate"]> {
+    return geometry
+        ? {
+              latitude: String(geometry.coordinates[1]),
+              longitude: String(geometry.coordinates[0]),
```

Comment:
I'd add full branch so we can declaritevely unpack the value: `const [latitude, longitude] = geometry.coordinates`. However, TS says the type of this is `coordinates: Coordinates | Coordinates[][]`. Something we are transforming to string, this needs some extra processing.

What to take from this? Both String(...) or .toString() accept any input, so we can introduce bugs. Let's create an abstraction for string->number (used before) and number->string for the components of coordinates.